de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/joiner/JoinerNodeModel.java

```diff
@@ -344,11 +354,27 @@ protected void loadJsonSetting() throws IOException, CanceledExecutionException
 
     if (flowVariables.containsKey("JoinRelations.json")) {
       String connectionString = flowVariables.get("JoinRelations.json").getStringValue();
-      nodeSettings.connections = MAPPER.readValue(connectionString, JoinRelation[].class);
+      try {
+        nodeSettings.connections = MAPPER.readValue(connectionString, JoinRelation[].class);
+      }catch(Exception ex) {
+        OldJoinerRelation[] oldRelations= MAPPER.readValue(connectionString, OldJoinerRelation[].class);
+        Function<OldJoinerRelation, JoinRelation> funcMigrateJoinerRelation= (OldJoinerRelation e)-> {return e.getNewJoinRelation();};
+        nodeSettings.connections  = Arrays.asList(oldRelations).stream().map(funcMigrateJoinerRelation
```

Comment:
We could stream the array directly with Arrays.stream instead of converting the array to a list and then stream it. See https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#stream-T:A-